### ISSUE TITLE
[General Usage]:  Question: provider login integration with Azure Active Directory

### ISSUE THREAD

**Issue by toni-moreno** [Snowflake-Labs/terraform-provider-snowflake#3206]
### Terraform CLI Version

1.5.0

### Terraform Provider Version

0.85.0

### Company Name

Can not put here sorry.

### Terraform Configuration

```terraform
We are working with terraform autenticating with snowflake with local user and password (generated as  environment vars before execution is done) , but  our security team is requesting us to integrate our terraform configurations with an service account provided from Azure AD. ( we have also snowflake autentication integrated with Azure AD SSO , but for interactive login)

I'm reviewing  autenticator information here (https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs#authenticator-2)  , and I can see that we can use a lot of different autentication modes. 

Question:

Which autenticator would be better to integrate with AzureAD in a non interactive way (no people will be there to write a password) ? There is any example anywhere to see how to configure this way?
```

### Category

category:provider_config

### Expected Behavior

something like that.

```
export SNOWFLAKE_USER="SERVIC_ACCOUNT_USER@MYORG.ORG"
export SNOWFLAKE_PASSWORD="<my_service_account_password>
```

### Actual Behavior

```
export SNOWFLAKE_USER="local_user"
export SNOWFLAKE_PASSWORD="local_password"
```

### Steps to Reproduce

No way to reproduce.

### How much impact is this issue causing?

Medium

**Comment by sfc-gh-jcieslak** [comment]
Hey @toni-moreno 
For most tested and basic non-interactive authentication method, we recommend authenticating by using private key ([here](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs#keypair-authentication-environment-variables)). If using AzureAD is a must, then most likely OAuth is your only way ([here](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs#oauth-access-token)).
cc: @sfc-gh-jmichalak

**Comment by sfc-gh-jmichalak** [comment]
That's correct with one caveat: `private_key_path` is currently deprecated and `private_key` should be used instead, optionally with [file](https://developer.hashicorp.com/terraform/language/functions/file) function. We'll update the linked documentation.

Please also read https://docs.snowflake.com/en/user-guide/oauth-azure

**Comment by toni-moreno** [comment]
Thank you  @sfc-gh-jcieslak  and @sfc-gh-jmichalak  for the fast response.

@sfc-gh-jcieslak  if I can not justify a better (secure) way to login  the AzureAD will be a must.   I've reviewed documentation  and I have some doubts on how to implement this autentication method I our terraform scripts. ( I'm a  bit ignorant on oauth for automation tools)
1) it seems the only way to have a more than 10 minutes  login is  configure   [oauth refresh token](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs#oauth-refresh-token),  do you know how to query on my snowflake account needed parameters? 

2) reviewing how oauth works  [here](https://docs.snowflake.com/en/user-guide/oauth-snowflake-overview)  it seems like this authentication way will need user interaction , but we need a way to automate our snowflake platform in a non interactive way.  Could you confirm us that Oauth could be non-interactive ?

thank you very much.

**Comment by sfc-gh-jcieslak** [comment]
Closing due to long inactivity.
